### PR TITLE
Improve TBC and general Rollbar error reporting

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #4287
Closes: #4163

## Description of Changes
- Removes the duplicate call to Rollbar in `setupErrorHandlers.ts`
- If the original error is passed to `ServerError` then it's message (if it exists) is passed to Rollbar as context
- Added the original error to `ServerError` in `/tokenBalance` route.
- Added the `Request` object to all Rollbar error reporting. This is very helpful as it allows us to track errors by user in Rollbar. It also allows us to see the request parameters at the time of the error.
- Updated the `validateTopicThreshold` function so that it throws if an error occurs rather than returning false. This allows the error to be propagated up to the route level so that the error is logged and reported to Rollbar.

## Test Plan
- Try posting, commenting, and reacting in a community with a token threshold without having enough tokens e.g. (Osmosis under 'Ministry of Marketing' topic). Have the console open and ensure that all errors returned are 400 hundred errors. The failed requests on the network tab should look like this:
  ```
  { status: 400, result: 'Insufficient Token Balance' }
  ```
- Do the same thing but this time with enough tokens (e.g. Osmosis under the 'General' topic where the threshold is 0) and ensure there are no logged errors (relating to TBC) or failed network requests.

## Deployment Plan

1. Normal Deployment

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/server/routes/tokenBalance.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ const tokenBalance = async (
     } else if (e.message === FetchTokenBalanceErrors.UnsupportedContractType) {
       throw new AppError(FetchTokenBalanceErrors.UnsupportedContractType);
     } else {
-      throw new ServerError(Errors.QueryFailed);
+      throw new ServerError(Errors.QueryFailed, e);
```
Comment (jnaviask) [inline packages/commonwealth/server/routes/tokenBalance.ts:66]: What does it mean to add the second error arg here?

**Review by jnaviask — CHANGES_REQUESTED** [review]
Need to add validateTopicThreshold try/catch in the server_threads_controller too.

File: packages/commonwealth/server/routes/tokenBalance.ts (hunk 1 above)
Comment (CowMuon) [reply packages/commonwealth/server/routes/tokenBalance.ts:66]: Isn't that so that the `underlyingError` prop on `ServerError` isn't undefined?

Comment (timolegros) [reply packages/commonwealth/server/routes/tokenBalance.ts:66]: Exactly:
```
export class ServerError extends Error {
  status: number;
  // Optionally include the original error that was thrown
  error: Error;
  constructor(message: string, error?: Error) {
    super(message);
    this.status = 500;
    this.name = 'ServerError';
    this.error = error;
  }
}
```

**Comment by timolegros** [comment]
> Need to add validateTopicThreshold try/catch in the server_threads_controller too.

Conflicts resolved.

Try-catch has been added. As discussed I created #4457 to address the broken conditionals which are out of the scope of this PR.

**Review by jnaviask — APPROVED** [review]

**Pull request merged**